src/completions.jl

```diff
@@ -13,6 +11,12 @@ function completion_is(ci::CompletionItem, ckind::Symbol)
         (ci.labelDetails.description === "argument" && ckind === :local)
 end
 
+function get_sort_text(ckind::Symbol, offset::Int)
+    # show non-global completions first in the list
+    n = ckind === :global ? (10000000 + offset) : offset
+    lpad(n, 8, '0')
+end
```

Comment:
This function includes `lpad` string operation and seems to have a small allocation cost every time it's called. So, maybe an approach with pre-allocation like below would be a bit better for performance. Thinking realistically, sorting 1000 candidates would be enough:
```suggestion
let sort_texts = Dict{Int, String}()
    for i = 1:1000
        sort_texts[i] = lpad(i, 4, '0')
    end
    max_sort_text, _ = maximum(sort_texts)
    global function get_sort_text(ckind::Symbol, offset::Int)
        if ckind === :global
            return max_sort_text
        end
        return get(sort_texts, offset, max_sort_text)
    end
end
```

Reply:
Thanks!  We could probably also limit the number of completions we send and set `isIncomplete = true`. I don't think anyone is going to scroll through 1000 completions before typing more characters in testing or real use.